realm/realm-library/src/objectServer/java/io/realm/internal/network/AuthServerResponse.java

```diff
@@ -63,7 +63,14 @@ public static ObjectServerError createError(String response, int httpErrorCode)
             JSONObject obj = new JSONObject(response);
             String title = obj.optString("title", null);
             String hint = obj.optString("hint", null);
-            ErrorCode errorCode = ErrorCode.fromInt(obj.optInt("code", -1));
+            ErrorCode errorCode;
+            if (obj.has("code")) {
+                errorCode = ErrorCode.fromInt(obj.getInt("code"));
+            } else if (obj.has("status")) {
```

Comment:
this looks bad :( are `code` and `status` the same thing? they all indicate the error code?

Reply:
Yes. https://github.com/realm/ros/issues/311 is tracking it